File: aspn-py/src/aspn23/type_satnav_time.py
"""
This code is generated via firehose.
DO NOT hand edit code.  Make any changes required using the firehose repo instead.
"""

from dataclasses import dataclass
from enum import Enum
from typing import Optional

from numpy.typing import NDArray


class TypeSatnavTimeTimeReference(Enum):

    """
    Reference time system used to express the data in this message. In a multi-GNSS receiver
    (GPS/GLONASS/Galileo/QZSS/BeiDou) all pseudorange observations must refer to one receiver clock
    only. The receiver clock time of the measurement is the receiver clock time of the received signals.
    It is identical for the phase and range measurements and is identical for all satellites observed in
    a given epoch.
    """

    """
    GPS system time.
    """
    TIME_GPS = 0

    """
    Galileo system time.
    """
    TIME_GALILEO = 1

    """
    BeiDou system time.
    """
    TIME_BEIDOU = 2

    """
    GLONASS system time.
    """
    TIME_GLONASS = 3


@dataclass
class TypeSatnavTime:
    """
    Satellite system time

    ### Attributes

    week_number - int:
            Full Week number since zero epoch at which the data provided in this message is
            valid, expressed in time system defined by time_reference enum (below).

    seconds_of_week - float:
            Seconds since start of current week at which the data provided in this message is
            valid, expressed in time system defined by time_reference enum (below).

    time_reference - TypeSatnavTimeTimeReference:
            Reference time system used to express the data in this message. In a multi-GNSS
            receiver (GPS/GLONASS/Galileo/QZSS/BeiDou) all pseudorange observations must refer to
            one receiver clock only. The receiver clock time of the measurement is the receiver
            clock time of the received signals. It is identical for the phase and range measurements
            and is identical for all satellites observed in a given epoch.
    """

    week_number: int
    seconds_of_week: float
    time_reference: TypeSatnavTimeTimeReference

File: aspn-c/src/aspn23/TypeSatnavTime.c

/*
 * This code is generated via firehose.
 * DO NOT hand edit code.  Make any changes required using the firehose repo instead
 */

#include "TypeSatnavTime.h"

Aspn23TypeSatnavTime* ASPN_NULLABLE
aspn23_type_satnav_time_new(uint16_t week_number,
                            double seconds_of_week,
                            enum Aspn23TypeSatnavTimeTimeReference time_reference) {
	Aspn23TypeSatnavTime* self = (struct Aspn23TypeSatnavTime*)malloc(sizeof(Aspn23TypeSatnavTime));
	if (NULL == self) return NULL;
	self->week_number     = week_number;
	self->seconds_of_week = seconds_of_week;
	self->time_reference  = time_reference;
	return self;
}

Aspn23TypeSatnavTime* ASPN_NULLABLE aspn23_type_satnav_time_copy(Aspn23TypeSatnavTime* input) {

	Aspn23TypeSatnavTime* out = aspn23_type_satnav_time_new(
	    input->week_number, input->seconds_of_week, input->time_reference);

	return out;
}

void aspn23_type_satnav_time_free(void* pointer) {
	Aspn23TypeSatnavTime* self = (Aspn23TypeSatnavTime*)pointer;
	if (NULL == self) return;
	aspn23_type_satnav_time_free_members(self);
	free(self);
}

void aspn23_type_satnav_time_free_members(Aspn23TypeSatnavTime* self) {
	if (NULL == self) return;
}
